From 0726127285413829f58618b5b82fb3e2da0c3a74 Mon Sep 17 00:00:00 2001
From: Andy Shaw <andy.shaw@digia.com>
Date: Tue, 8 Oct 2013 15:45:42 +0200
Subject: [PATCH] Ensure lastPixel.y is also initalized to -1 when necessary

Task-number: QTBUG-31579

Upstream-Status: Accepted [https://codereview.qt-project.org/#change,67624]
Signed-off-by: Jonathan Liu <net147@gmail.com>

Change-Id: Ia4916ac0d384ea523bbfad0dafbc90044da7ec3e
Reviewed-by: Gunnar Sletta <gunnar.sletta@digia.com>
(cherry picked from qtbase/f0b950144dde85b65df0c3cf83f7c8df0e94bbbf)
---
 src/gui/painting/qcosmeticstroker.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gui/painting/qcosmeticstroker.cpp b/src/gui/painting/qcosmeticstroker.cpp
index 4413170..36616d4 100644
--- a/src/gui/painting/qcosmeticstroker.cpp
+++ b/src/gui/painting/qcosmeticstroker.cpp
@@ -305,6 +305,7 @@ void QCosmeticStroker::setup()
     ymax = deviceRect.bottom() + 2;
 
     lastPixel.x = -1;
+    lastPixel.y = -1;
 }
 
 // returns true if the whole line gets clipped away
@@ -533,6 +534,7 @@ void QCosmeticStroker::drawPath(const QVectorPath &path)
             QPointF p = QPointF(points[0], points[1]) * state->matrix;
             patternOffset = state->lastPen.dashOffset()*64;
             lastPixel.x = -1;
+            lastPixel.y = -1;
 
             bool closed;
             const QPainterPath::ElementType *e = subPath(type, end, points, &closed);
@@ -587,6 +589,7 @@ void QCosmeticStroker::drawPath(const QVectorPath &path)
         QPointF movedTo = p;
         patternOffset = state->lastPen.dashOffset()*64;
         lastPixel.x = -1;
+        lastPixel.y = -1;
 
         const qreal *begin = points;
         const qreal *end = points + 2*path.elementCount();
-- 
1.8.4

